### PR TITLE
Upgrade Decidim to 0.27

### PR DESCRIPTION
**Pull request by NicoleeSo13**
This PR upgrades the Granada Dipgra App to Decidim v0.27.2.

Release: https://github.com/decidim/decidim/releases :
- 0.27.2 : https://github.com/decidim/decidim/releases/tag/v0.27.2
- 0.27.1 : https://github.com/decidim/decidim/releases/tag/v0.27.1
- 0.27.0: https://github.com/decidim/decidim/releases/tag/v0.27.0 

Note that many Decidim configuration has moved from the `config/initializers/decidim.rb` to ENV variables. See https://github.com/decidim/decidim/pull/8725 for a full list of ENV vars.

To execute after deploy:
https://github.com/decidim/decidim/blob/release/0.27-stable/CHANGELOG.md#3-one-time-actions

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,10 +30,11 @@ gem "deface"
...
 # gem "whenever"
-# elsif deploying to Heroku
+# elsif deploying to a PaaS like Heroku
 gem "redis"
 gem "sidekiq"
 group :production do
+  gem "aws-sdk-s3", require: false
   gem "fog-aws"
   gem "rack-ssl-enforcer"
   gem "rails_12factor"
```
Comment (tramuntanal) [inline Gemfile]: Again, this two blocs must be inversely commented. This application is deployed in a traditional server.

File: decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_handler.rb [hunk 2]
```diff
@@ -54,11 +54,13 @@ def census_for_user
     return @census_for_user if defined? @census_for_user
     return unless organization
 
-    @service = DipgraCensusAuthorization.new(DipgraCensusAuthorizationConfig.api_config(organization))
+    username = Rails.application.secrets.dipgra_census[:username]
+    password = Rails.application.secrets.dipgra_census[:password]
+    @service = DipgraCensusAuthorization.new(username: "#{organization.ine_code}#{username}", password:, organization:)
```
Comment (tramuntanal) [inline decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_handler.rb]: why this change?

File: decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb [hunk 3]
```diff
@@ -4,22 +4,24 @@
...
 # rubocop:disable RSpec/MultipleMemoizedHelpers
 RSpec.describe DipgraCensusAuthorization do
-  let(:organization) { create(:organization) }
-  let(:subject) do
+  subject do
     allow(Rails.application.secrets).to receive(:dipgra_census).and_return(JSON.parse({
       username: "username",
       password: "password",
-      url: "http://dipgra.ws/services/Ci",
+      url: "http://80//dipgra.ws/services/Ci",
```
Comment (tramuntanal) [inline decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb]: why this change?

File: decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb [hunk 4]
```diff
@@ -30,7 +32,7 @@
     let(:rs_birthdate) { "20000101000000" }
 
     before do
-      stub_request(:post, "http://dipgra.ws/services/Ci")
+      stub_request(:post, "http://80/dipgra.ws/services/Ci")
```
Comment (tramuntanal) [inline decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb]: again, why this change?

**Review by tramuntanal — CHANGES_REQUESTED** [review]
Good job @NicoleSo13 !

Please check the comments, I have some questions because I don't understand the changes. Maybe we can comment them offline.

**Comment by laurajaime** [comment]
@NicoleSo13 can you check @tramuntanal comments and rerun the tests after? thanks!

File: decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_handler.rb (hunk 2 above)
Comment (NicoleeSo13) [reply decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_handler.rb]: Because after update from Ruby 2.7 to Ruby 3.1 old syntax stopped working, so I had to rewrite some partes of this module

File: decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb (hunk 3 above)
Comment (NicoleeSo13) [reply decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb]: It was my attempt to solve
```
1st Try error in ./spec/services/dipgra_census_authorization_spec.rb:49:
Absolute URI missing hierarchical segment: 'http://:80'
```
in tests

File: decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_handler.rb (hunk 2 above)
Comment (laurajaime) [reply decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_handler.rb]: what was the mistake? it's weird that it doesn't work

File: decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb (hunk 3 above)
Comment (laurajaime) [reply decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb]: you must remove 80//

Comment (NicoleSo13) [reply decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb]: But it's removed already

File: decidim-dipgra_census_authorization/Gemfile
Comment (laurajaime) [inline decidim-dipgra_census_authorization/Gemfile:1]: the changes in this file are not necessary.

File: .github/workflows/ci_app.yml [hunk 5]
```diff
@@ -30,7 +30,8 @@ jobs:
         image: redis:3.2-alpine
         ports: ["6379:6379"]
     env:
-      DB_DATABASE: app-test
+      DB_TEST_DATABASE: app-test
+      DB_DATABASE: app
```
Comment (laurajaime) [inline .github/workflows/ci_app.yml:34]: which one is correct?

File: .node_version
Comment (laurajaime) [inline .node_version:1]: we must add `.node-version` with `16.9.1`

File: Gemfile [hunk 6]
```diff
@@ -4,13 +4,15 @@ source "https://rubygems.org"
 
 ruby RUBY_VERSION
 
-DECIDIM_VERSION = { git: "https://github.com/CodiTramuntana/decidim.git", branch: "release/0.26-stable" }.freeze
+DECIDIM_VERSION = { git: "https://github.com/CodiTramuntana/decidim.git", branch: "release/0.27-stable" }.freeze
 
 gem "decidim", DECIDIM_VERSION
 gem "decidim-dipgra_census_authorization", path: "decidim-dipgra_census_authorization"
+gem "decidim-file_authorization_handler", git: "https://github.com/CodiTramuntana/decidim-file_authorization_handler.git", tag: "v0.27.1.1"
+gem "decidim-term_customizer", git: "https://github.com/mainio/decidim-module-term_customizer.git", branch: "master"
```
Comment (laurajaime) [inline Gemfile]:
```suggestion
gem "decidim-term_customizer", git: "https://github.com/mainio/decidim-module-term_customizer.git"
```

File: Gemfile [hunk 7]
```diff
@@ -24,18 +26,15 @@ gem "openssl"
 
 gem "deface"
 
-# if deploying to a dedicated server
-# gem "daemons"
-# gem "delayed_job_active_record"
-# gem "whenever"
```
Comment (laurajaime) [inline Gemfile:30]: we must uncomment that and comment heroku deploying gems.

File: config/initializers/decidim.rb [hunk 8]
```diff
@@ -1,25 +1,66 @@
...
-  config.default_locale = :es
-  config.available_locales = [:en, :es]
+  # Sets the list of available locales for the whole application.
+  #
+  # When an organization is created through the System area, system admins will
+  # be able to choose the available languages for that organization. That list
+  # of languages will be equal or a subset of the list in this file.
+  # config.available_locales = Rails.application.secrets.decidim[:available_locales].presence || [:en]
+  # Or block set it up manually and prevent ENV manipulation:
+  config.available_locales = %w(ca en es)
```
Comment (laurajaime) [inline config/initializers/decidim.rb]: we must remove ca

```suggestion
  config.available_locales = %w(en es)
```

File: decidim-dipgra_census_authorization/Gemfile
Comment (laurajaime) [inline decidim-dipgra_census_authorization/Gemfile:1]: This changes are not required here. Can you undo them?

File: decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_config.rb [hunk 9]
```diff
@@ -24,13 +24,5 @@ def password
...
       Rails.application.secrets.dipgra_census[:public_key]
     end
-
-    def api_config(organization)
-      {
-        username: "#{organization.ine_code}#{username}",
-        password: password,
-        organization: organization
-      }
-    end
```
Comment (laurajaime) [inline decidim-dipgra_census_authorization/app/services/dipgra_census_authorization_config.rb]: Can you recover that method? I don't understand why you remove it.

File: decidim-dipgra_census_authorization/decidim-dipgra_census_authorization.gemspec [hunk 10]
```diff
@@ -26,4 +26,5 @@ Gem::Specification.new do |s|
 
   s.add_development_dependency "decidim-dev", DECIDIM_VERSION
   s.add_development_dependency "faker"
+  s.metadata["rubygems_mfa_required"] = "true"
```
Comment (laurajaime) [inline decidim-dipgra_census_authorization/decidim-dipgra_census_authorization.gemspec:28]: what is this for?

File: decidim-dipgra_census_authorization/lib/decidim/dipgra_census_authorization/version.rb [hunk 11]
```diff
@@ -2,6 +2,6 @@
 
 module Decidim
   module DipgraCensusAuthorization
-    VERSION = "0.26.2"
+    VERSION = "0.27"
```
Comment (laurajaime) [inline decidim-dipgra_census_authorization/lib/decidim/dipgra_census_authorization/version.rb]: I don't know if we must set `0.27.0`... cc/ @tramuntanal

File: decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb
Comment (laurajaime) [inline decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb:1]: we review the tests when we see the other changes

**Review by laurajaime — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 7 above)
Comment (NicoleeSo13) [reply Gemfile:30]: Look at the complete file, all that lines are uncommented but was duplicated so that's why I removed them. All those three gems were already in the Gemfile. I have commented Heroku deploying gems too

File: decidim-dipgra_census_authorization/decidim-dipgra_census_authorization.gemspec (hunk 10 above)
Comment (NicoleeSo13) [reply decidim-dipgra_census_authorization/decidim-dipgra_census_authorization.gemspec:28]: Rubocop: C: [Correctable] Gemspec/RequireMFA: metadata['rubygems_mfa_required'] must be set to 'true'.

File: decidim-dipgra_census_authorization/Gemfile
Comment (NicoleeSo13) [reply decidim-dipgra_census_authorization/Gemfile:1]: Without that change with "psych <4" the pipeline will fail on generate dipgra_census_authorization_test_app

File: decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb (hunk 4 above)
Comment (NicoleeSo13) [reply decidim-dipgra_census_authorization/spec/services/dipgra_census_authorization_spec.rb]: First ones was due to Rubocop, the ones with http://80 is already reverted

File: .github/workflows/ci_app.yml (hunk 5 above)
Comment (NicoleeSo13) [reply .github/workflows/ci_app.yml:34]: I'm not sure about it, but I think current version makes more sense becasuse we have both DB_TEST_DATABASE and DB_DATABASE in application.yml

File: Gemfile (hunk 6 above)
Comment (NicoleeSo13) [reply Gemfile]: Corrected

File: Gemfile (hunk 1 above)
Comment (NicoleeSo13) [reply Gemfile]: I have got them inversely commented, but as some gems were duplicated after uncommenting them, I have decided to remove duplicated entries

File: Gemfile (hunk 7 above)
Comment (laurajaime) [inline Gemfile:30]: this gems must be uncommented

**Review by laurajaime — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_app.yml (hunk 5 above)
Comment (laurajaime) [reply .github/workflows/ci_app.yml:34]: can you try removing `DB_DATABASE` and check if CI is still works?

File: Gemfile (hunk 7 above)
Comment (NicoleeSo13) [reply Gemfile:30]: All these three gems are already in the Gemfile. These were duplicates

File: .github/workflows/ci_app.yml (hunk 5 above)
Comment (NicoleeSo13) [reply .github/workflows/ci_app.yml:34]: CI stops working after removing either DB_DATABASE or DB_TEST_DATABASE

**Review by laurajaime — APPROVED** [review]
